### PR TITLE
Update AdvantageScope docs

### PR DESCRIPTION
**Pull request by jwbonner**
These updates reflect AdvantageScope's inclusion in the WPILib installer for 2024, with a dedicated page and references from relevant locations in the rest of the docs. The idea is to keep the main AdvantageScope docs separate but make the tool more discoverable.

I think the main questions here relate to organization. The AdvantageScope page is under "Telemetry" in this version, but I could see arguments for putting it under "Dashboards" (with Glass, which is also a programmer's tool) or "Software Tools" (with OutlineViewer). I also added AdvantageScope references to the pages for Field2d, Mechanism2d, and plots; these are currently organized under "Glass", but the Field2d and plots pages are also part of "Robot Simulation" where I think the inclusion of AdvantageScope makes more sense. There's a bit of a disconnect now between those features and Glass, but maybe keeping AdvantageScope as a side reference on those pages is an OK solution? I'm open to suggestions.

I'm opening this as a draft for now since the explanation for how to open AdvantageScope won't actually work until it's included in the tools menu and file associations are added for Windows.

### CODE REVIEW

**Review by jasondaming — CHANGES_REQUESTED** [review]
Good stuff I think we should merge this ASAP with a couple changes.

We can defer this to a different PR if you want but I really thing we should have a section on publishing swerve states (I don't remember the current state of structs?).  For troubleshooting and getting started with Swerve this is one of the most helpful things teams can do

**Comment by jwbonner** [comment]
A few changes:

- I split the DataLog page in two to address #2422, with sections for AdvantageScope and the DataLogTool. Another section for the DS could be added there in the future.
- Based on the changes in #2428, I moved the main AdvantageScope page to "Dashboards". I think this fits better since the new data log download page mentions AdvantageScope already.
- I added a section to the swerve drive kinematics page for AdvantageScope, which seemed like the most obvious place to talk about swerve visualization since the `SwerveModuleState` class is introduced on that page. There is example code for publishing the states to NT in Java and C++. Python should also be included, I just haven't had a chance to figure out how struct publishing works for that.

**Comment by jasondaming** [comment]
I am not sure struct publishing is done for python.  At least I couldn't [find it](https://robotpy.readthedocs.io/projects/pyntcore/en/stable/ntcore.html).

@virtuald do you know how we would do this?

**Comment by virtuald** [comment]
It is not implemented yet: https://github.com/robotpy/mostrobotpy/issues/5

**Comment by jasondaming** [comment]
I don't think there is anything preventing this from being merged with a little cleanup.

File: source/docs/software/telemetry/advantagescope.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+- Specialized displays for joysticks, swerve module states, and console text
+- Analysis of numeric fields using histograms and statistical measures
+- Multiple export options, including CSV and WPILib data logs
+
+.. image:: images/advantagescope.png
+   :alt: Screenshot of an AdvantageScope window displaying a line graph, a list of fields, and a series of miscellaneous tabs.
+
+In Visual Studio Code, press :kbd:`Ctrl+Shift+P` and type ``WPILib`` or click the WPILib logo in the top right to launch the WPILib Command Palette. Select :guilabel:`Start Tool`, then select :guilabel:`AdvantageScope`. You can also open any supported log file in AdvantageScope using a standard file browser.
+
+.. note:: Detailed documentation for AdvantageScope can be found `here <https://github.com/Mechanical-Advantage/AdvantageScope/blob/main/docs/INDEX.md>`__. It is also available offline by clicking the book icon in the tab bar.
```
[the remark this answers is not in the document]
Comment (jwbonner) [reply source/docs/software/telemetry/advantagescope.rst]: This note is now moved near the top, above the screenshot and list of features.

**Review by jasondaming — APPROVED** [review]

**Comment by virtuald** [comment]
FWIW python struct publishing is now merged: https://github.com/robotpy/mostrobotpy/pull/50

**Pull request merged**